Sources/Xcodeproj/pbxproj().swift

```diff
@@ -530,6 +530,29 @@ func xcodeProject(
             }
         }
     }
+    
+    // Create an aggregate target for every product for which there isn't already
+    // a target with the same name.
+    let targetNames: Set<String> = Set(modulesToTargets.values.map{ $0.name })
+    for product in graph.products {
+        // Go on to next product if we already have a target with the same name.
+        if targetNames.contains(product.name) { continue }
+        // Otherwise, create an aggreate target.
+        let target = project.addTarget(productType: nil, name: product.name)
```

Comment:
Nope, I don't remember changing anything non trivial in Xcode project generation. I agree with your point that we should only create the targets needed for the products (and plus all the root targets). We already have this logic wired up in the new build plan, we should try to use that here, I think.